tests/test_scheduler.py

```diff
@@ -73,11 +75,15 @@ def model(sample, t, *args):
         return model
 
     def check_over_configs(self, time_step=0, **config):
+        print(config)
         kwargs = dict(self.forward_default_kwargs)
 
         num_inference_steps = kwargs.pop("num_inference_steps", None)
 
         for scheduler_class in self.scheduler_classes:
+            if scheduler_class in (EulerAncestralDiscreteScheduler, EulerDiscreteScheduler, LMSDiscreteScheduler):
+                time_step = float(time_step)
```

Comment:
Need to convert this to float as the new euler schedulers raise error for int timesteps cc @patrickvonplaten

Reply:
It sounds a bit strange to me that the types are different across schedulers. I think it's ok for now, but maybe we need to standardize somehow. Maybe use a function `timestep_for_index` that all must implement? I don't know, maybe worth revisiting at some point.

Reply:
Good point, let's open an issue for this. To me also this seems a bit strange, as we are calling it Discreet scheduler but passing float timesteps. cc @anton-l @patrickvonplaten

Reply:
I see this makes sense! I'd say we update DDIM, DDPM and PNDM to float time steps (even if they are ints) to be able to delete this line in a next PR